sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2FunctionSuite.scala

```diff
@@ -466,6 +526,60 @@ class DataSourceV2FunctionSuite extends DatasourceV2SQLBase {
     override def produceResult(state: (Long, Long)): Long = state._1 / state._2
   }
 
+  object UnboundDecimalAverage extends UnboundFunction {
+    override def name(): String = "favg"
+
+    override def bind(inputType: StructType): BoundFunction = {
+      if (inputType.fields.length > 1) {
+        throw new UnsupportedOperationException("Too many arguments")
+      }
+
+      // put interval type here for testing purpose
+      inputType.fields(0).dataType match {
+        case _: NumericType | _: DayTimeIntervalType => DecimalAverage
+        case dataType =>
+          throw new UnsupportedOperationException(s"Unsupported input type: $dataType")
+      }
+    }
+
+    override def description(): String =
+      """iavg: produces an average using decimal division, ignoring nulls
+        |  iavg(integral) -> decimal
+        |  iavg(float) -> decimal
+        |  iavg(decimal) -> decimal""".stripMargin
+  }
+
+  object DecimalAverage extends AggregateFunction[(Decimal, Int), Decimal] {
+    val PRECISION: Int = 15
+    val SCALE: Int = 5
```

Comment:
Sorry, but may I ask where these (precision 15, scale 5) came from? I might miss the context, but this looks like not non-conventional values in Apache Spark codebase. Is it intentionally chosen for some purpose?

Reply:
sorry this is a little confusing - I was using this to test some negative case like passing `long` which is `decimal(20, 0)` to this `decimal(15, 5)` which won't work without losing precision. But, I no longer need this now. Will remove.